pure-selenium/src/main/java/org/example/pages/InventoryPage.java

```diff
@@ -1,44 +1,97 @@
 package org.example.pages;
 
+import org.example.components.ProductListItemComponent;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
-import org.openqa.selenium.support.PageFactory;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
 
+import java.time.Duration;
+import java.util.ArrayList;
 import java.util.List;
+import org.example.enums.SortOption;
+import org.openqa.selenium.support.ui.WebDriverWait;
 
-public class InventoryPage {
+public class InventoryPage extends AbstractPage{
 
-    private WebDriver driver;
-    private static final Logger logger = LoggerFactory.getLogger(InventoryPage.class);
+    @FindBy(xpath = "//span[contains(@data-test,'title')]")
+    private WebElement pageTitle;
 
+    @FindBy(id = "react-burger-menu-btn")
+    private WebElement menuButton;
 
-    @FindBy(css = ".inventory_item_name")
-    private List<WebElement> productTitles;
+    @FindBy(id = "logout_sidebar_link")
+    private WebElement logoutLink;
 
-    @FindBy(css = "[data-test='product_sort_container']")
+    @FindBy(className = "shopping_cart_badge")
+    private WebElement cartBadge;
+
+    @FindBy(className = "shopping_cart_link")
+    private WebElement cartIcon;
+
+    @FindBy(className = "product_sort_container")
     private WebElement sortDropdown;
 
+    @FindBy(className = "inventory_item")
+    private List<WebElement> productElements;
+
     public InventoryPage(WebDriver driver) {
-        this.driver = driver;
-        PageFactory.initElements(driver, this);
+        super(driver);
     }
 
-//    public List<WebElement> getProductTitles() {
-//        return productTitles;
-//    }
+    public void sortBy(SortOption option) {
+        Select select = new Select(sortDropdown);
+        select.selectByVisibleText(option.getVisibleText());
+    }
 
-    public void printAllProductTitles() {
-        if (productTitles.isEmpty()) {
-            logger.warn("No products found!");
-        } else {
-            logger.info("Products:");
-            for (WebElement title : productTitles) {
-                logger.info("- {}", title.getText());
+    public List<Double> getDisplayedPrices() {
+        List<Double> prices = new ArrayList<>();
+        for (WebElement element : productElements) {
+            ProductListItemComponent product = new ProductListItemComponent(driver, element);
+            String priceText = product.getPrice().replace("$", "");
+            prices.add(Double.parseDouble(priceText));
```

Comment:
In Carina it looks like,
@FindBy(xpath = "//div[@class='inventory_item']")
    private List<ProductListItemComponent> productListItems;
    
public List<Double> getDisplayedPrices() {
    List<Double> prices = new ArrayList<>();
    for (ProductListItemComponent product : productListItems) {
        prices.add(product.getPriceAsDouble());
    }
    return prices;
}